komodo/cli.py

```diff
@@ -2,158 +2,206 @@
 
 import argparse
 import os
-import shutil
 import sys
+
 import yaml as yml
-import logging
-from textwrap import dedent
 
 import komodo
-from komodo.shim import create_shims
+import komodo.local as local
+import komodo.switch as switch
+from komodo.data import Data
 
 
 def _main(args):
     args.prefix = os.path.abspath(args.prefix)
 
+    data = Data(extra_data_dirs=args.extra_data_dirs)
+
     if args.download or (not args.build and not args.install):
-        komodo.fetch(args.pkgs, args.repo, outdir = args.cache,
-                     pip = args.pip,
-                     git = args.git)
+        komodo.fetch(
+            args.pkgs, args.repo, outdir=args.cache, pip=args.pip, git=args.git
+        )
 
     if args.download and not args.build:
         sys.exit(0)
 
     # append root to the temporary build dir, as we want a named root/
     # directory as the distribution root, organised under the distribution name
     # (release)
-    tmp_prefix = os.path.join(os.path.join(args.prefix), args.release, 'root')
+    tmp_prefix = os.path.join(os.path.join(args.prefix), args.release, "root")
     fakeroot = os.path.abspath(args.release)
     if args.build or not args.install:
-        komodo.make(args.pkgs, args.repo, prefix   = tmp_prefix,
-                                         dlprefix = args.cache,
-                                         builddir = args.tmp,
-                                         jobs     = args.jobs,
-                                         cmk      = args.cmake,
-                                         pip      = args.pip,
-                                         fakeroot = fakeroot)
-        komodo.shell('mv {} {}'.format(args.release + tmp_prefix, args.release))
-        komodo.shell('rmdir -p --ignore-fail-on-non-empty {}'.format(
-            os.path.dirname(args.release + tmp_prefix)))
+        komodo.make(
+            args.pkgs,
+            args.repo,
+            data,
+            prefix=tmp_prefix,
+            dlprefix=args.cache,
+            builddir=args.tmp,
+            jobs=args.jobs,
+            cmk=args.cmake,
+            pip=args.pip,
+            virtualenv=args.virtualenv,
+            fakeroot=fakeroot,
+        )
+        komodo.shell("mv {} {}".format(args.release + tmp_prefix, args.release))
+        komodo.shell(
+            "rmdir -p --ignore-fail-on-non-empty {}".format(
+                os.path.dirname(args.release + tmp_prefix)
+            )
+        )
 
     if args.build and not args.install:
         sys.exit(0)
 
     # create the enable script
-    for tmpl,target in [('enable.in','enable'), ('enable.csh.in', 'enable.csh')]:
-        if not os.path.exists(tmpl):
-            logging.warn('Could not find template %s, skipping.' % str(tmpl))
-            continue
+    for tmpl, target in [("enable.in", "enable"), ("enable.csh.in", "enable.csh")]:
+        # if not os.path.exists(tmpl):
```

Comment:
either comment back in or remove

Reply:
Right, we don't want to warn here, we want to fail hard. Will fix.